airbyte-ci/connectors/regression-testing/regression_testing/main.py

```diff
@@ -0,0 +1,163 @@
+import asyncio
+import logging
+import sys
+from typing import Dict, Optional
+
+import click
+import dagger
+from airbyte_protocol.models import ConfiguredAirbyteCatalog
+
+from .backends import BaseBackend, FileBackend
+from .connector_runner import ConnectorRunner, SecretDict
+from .utils import get_connector, get_connector_config, get_state
+
+logging.basicConfig(level=logging.INFO)
+
+logger = logging.getLogger(__name__)
+
+
+async def _main(
+    connector_name: str,
+    control_image_name: str,
+    target_image_name: str,
+    output_directory: str,
+    command: Optional[str],
+    config_path: Optional[str],
+    catalog_path: Optional[str],
+    state_path: Optional[str],
+):
+    if control_image_name in ("dev", "latest"):
+        control_image_name = f"airbyte/{connector_name}:{control_image_name}"
+
+    if target_image_name in ("dev", "latest"):
+        target_image_name = f"airbyte/{connector_name}:{target_image_name}"
+
+    # TODO: add backend options
+    if output_directory:
+        backend = FileBackend(output_directory)
+    else:
+        raise NotImplementedError(f"An output directory is required; only file backends are currently supported.")
+
+    config = get_connector_config(config_path)
+    catalog = ConfiguredAirbyteCatalog.parse_file(catalog_path) if catalog_path else None
+    state = get_state(state_path) if state_path else None
+
+    async with dagger.Connection(config=dagger.Config(log_output=sys.stderr)) as client:
+        control_container = await get_connector(client, control_image_name)
+        control_version = control_image_name.split(":")[-1]
+
+        target_container = await get_connector(client, target_image_name)
+        target_version = target_image_name.split(":")[-1]
+
+        # TODO: maybe use vcrpy to cache the response from the first round and use the cassette for the second round
+        #   (this may only make sense for syncs with an input state)
+        tasks = [
+            dispatch(
+                container,
+                backend,
+                f"{output_directory}/{version}",
+                command,
+                config,
+                catalog,
+                state,
+            ) for version, container in [(control_version, control_container), (target_version, target_container)]
+        ]
+        await asyncio.gather(*tasks)
+
+
+async def dispatch(
```

Comment:
Am I correct saying that the idea is to dispatch commands, and then perform assertions on the outputs in tests? 
So tests are not running commands directly, outputs are considered test fixtures? 
(I've nothing against this, just make sure I get your plan).